src/classes/HH_Container_LCTRL.cls

```diff
@@ -349,15 +349,22 @@ public with sharing class HH_Container_LCTRL {
             // allow potential duplicates when rules to warn are turned on.
             Database.DMLOptions dml = new Database.DMLOptions();
             dml.DuplicateRuleHeader.AllowSave = true;
+
             list<Database.SaveResult> listSR = Database.insert(listInsert, dml);
             for (Database.SaveResult sr : listSR) {
                 if (!sr.isSuccess())
                     throw new DmlException(sr.getErrors()[0].getMessage());
             }
+
             update listUpdate;
-            TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.flag.HH, false);
+
         } catch (Exception ex) {
+            Database.rollback(sp);
+
             throw new AuraHandledException(ex.getMessage());
+
+        } finally {
+            TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.flag.HH, false);
```

Comment:
I don't mind that you moved these into a finally block.  but I just want to confirm that since we are throwing an exception back to Aura and the client, that the transaction scope would have completed, and our state flags would have gone out of scope, no?

Reply:
Yes, that is correct. If a such method is called from a VF page, then changes applied to the static variables are reset when the method completes regardless if it is success or not. The issue is only applicable if the method is called from another one and the caller catches the exception and continues the transaction: static/instance/local (primitive data type or SObject) variables are not implicitly rolled back.
The reset of the static variable in the above method was not required even though the method is called from another one. The caller does not catch the exception and the transaction stops if the exception has been raised. I can revert back changes -- please let me know.

Reply:
the finally is fine.